### PR TITLE
backend-tasks: allow tasks to run more frequently than the default work check interval

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

Found that the default work check interval of 5s made it impossible run to tasks more often than that on a single node. Not sure we necessarily want exactly this but at the same time it feel like we should allow for tasks to be scheduled more frequently than 5s if desired.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/backend-tasks/src/tasks/TaskWorker.ts [hunk 1]
```diff
@@ -51,6 +51,10 @@ export class TaskWorker {
       `Task worker starting: ${this.taskId}, ${JSON.stringify(settings)}`,
     );
 
+    const cadence = Duration.fromISO(settings.cadence);
+    const workCheckFrequency =
+      cadence < this.workCheckFrequency ? cadence : this.workCheckFrequency;
```
Comment (freben) [inline packages/backend-tasks/src/tasks/TaskWorker.ts:56]: Can you really compare them like this? I'd assumed you needed to compare their respective `.as('milliseconds')` or so

File: packages/backend-tasks/src/tasks/TaskWorker.ts [hunk 2]
```diff
@@ -51,6 +51,10 @@ export class TaskWorker {
       `Task worker starting: ${this.taskId}, ${JSON.stringify(settings)}`,
     );
 
+    const cadence = Duration.fromISO(settings.cadence);
```
Comment (freben) [inline packages/backend-tasks/src/tasks/TaskWorker.ts:54]: I may be misremembering, but can't the cadence be a cron string too? (Not at computer now)

File: packages/backend-tasks/src/tasks/TaskWorker.ts (hunk 1 above)
Comment (Rugvip) [reply packages/backend-tasks/src/tasks/TaskWorker.ts]: This is apparently the intended way, I was searching for a `.compare(...)` or something like that 😅

File: packages/backend-tasks/src/tasks/TaskWorker.ts (hunk 2 above)
Comment (Rugvip) [reply packages/backend-tasks/src/tasks/TaskWorker.ts]: Yep, updated!

**Pull request merged**